src/openrct2/network/NetworkGroup.cpp

```diff
@@ -102,9 +102,9 @@ void NetworkGroup::Write(NetworkPacket &packet)
 {
     packet << Id;
     packet.WriteString(GetName().c_str());
-    for (size_t i = 0; i < ActionsAllowed.size(); i++)
+    for (auto &action : ActionsAllowed)
```

Comment:
`const auto &action`

Reply:
If I make it `const auto &action`, I get the following compiler error:

```
In file included from ../src/openrct2/network/NetworkGroup.h:22:0,
                 from ../src/openrct2/network/NetworkGroup.cpp:21:
../src/openrct2/network/NetworkPacket.h: In instantiation of ‘NetworkPacket& NetworkPacket::operator>>(T&) [with T = const unsigned char]’:
../src/openrct2/network/NetworkGroup.cpp:97:19:   required from here
../src/openrct2/network/NetworkPacket.h:53:19: error: assignment of read-only reference ‘value’
             value = 0;
             ~~~~~~^~~
../src/openrct2/network/NetworkPacket.h:57:19: error: assignment of read-only reference ‘value’
             value = ByteSwapBE(*((T *)&GetData()[BytesRead]));
             ~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
ninja: build stopped: subcommand failed.
```

Reply:
That's some horrid violation of C++

Reply:
Huh?, It worked for me

Reply:
I only said to do it for the second one.

Reply:
Only make this one const, the other one can't be const because there the values get's written to `action`. Here `action` is only only used for writing into the packet.